map_test.go

```diff
@@ -0,0 +1,127 @@
+package protobuf
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+	// for more human friendly hex dump output (first ... last 3 bytes):
+	// goprotobuf "github.com/golang/protobuf/proto"
+)
+
+type Inner struct {
+	Id   int32
+	Name string
+}
+
+type FloatingPoint struct {
+	F *float64 `protobuf:"1,req"`
+}
+
+type MessageWithMap struct {
+	NameMapping   map[uint32]string // = 1, required
+	ByteMapping   map[bool][]byte
+	MsgMapping    map[int64]*FloatingPoint
+	StrToStr      map[string]string
+	StructMapping map[string]*Inner
+}
+
+func TestMapFieldEncode(t *testing.T) {
```

Comment:
Why not put small comments in front of each test, specifying which part of the implementation is tested?